src/App.js

```diff
@@ -1,16 +1,44 @@
 import React from 'react';
 import './App.css';
+import { useState } from 'react';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const updateLike = () => {
+    let likeCount = 0;
+
+    for (const entry of entries) {
+      if (entry.liked === true) {
+        likeCount += 1;
+      }
+    }
+    return likeCount;
+  };
+
+  const toggleHeart = (id) => {
+    const newEntries = [];
+
+    for (const entry of entries) {
+      const newEntry = { ...entry };
+      if (newEntry.id === id) {
+        newEntry.liked = !newEntry.liked;
+      }
+      newEntries.push(newEntry);
+    }
+    setEntries(newEntries);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat with Estragon and Vladimir</h1>
+        <div> {updateLike()}❤️s</div>
```

Comment:
As mentioned in my comment on Learn, this causes a test failure because our tests are expecting "3 ❤️s" with a space in between the number and the heart, while your code produces "3❤️s".

But this is a relatively minor issue, obviously.